pkg/storage/driver/sql.go

```diff
@@ -0,0 +1,361 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package driver // import "helm.sh/helm/v3/pkg/storage/driver"
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	migrate "github.com/rubenv/sql-migrate"
+
+	// Import pq for postgres dialect
+	_ "github.com/lib/pq"
+
+	storageerrors "k8s.io/helm/pkg/storage/errors"
+
+	rspb "helm.sh/helm/v3/pkg/release"
+)
+
+var _ Driver = (*SQL)(nil)
+
+var labelMap = map[string]struct{}{
+	"modifiedAt": {},
+	"createdAt":  {},
+	"version":    {},
+	"status":     {},
+	"owner":      {},
+	"name":       {},
+}
+
+var supportedSQLDialects = map[string]struct{}{
+	"postgres": {},
+}
+
+// SQLDriverName is the string name of this driver.
+const SQLDriverName = "SQL"
+
+// SQL is the sql storage driver implementation.
+type SQL struct {
+	db        *sqlx.DB
+	namespace string
+
+	Log func(string, ...interface{})
+}
+
+// Name returns the name of the driver.
+func (s *SQL) Name() string {
+	return SQLDriverName
+}
+
+func (s *SQL) ensureDBSetup() error {
+	// Populate the database with the relations we need if they don't exist yet
+	migrations := &migrate.MemoryMigrationSource{
+		Migrations: []*migrate.Migration{
+			{
+				Id: "init",
+				Up: []string{
+					`
+						CREATE TABLE releases_v3 (
```

Comment:
Can we call the table `release.v1` ?
This would correspond to the type used for Secret/Configmap storage: https://github.com/helm/helm/blob/master/pkg/storage/driver/secrets.go#L236

Reply:
@hickeyma postgres uses dot notation to access databases, schemes and table names. Therefore it is not allowed to use them for table/column names. There are workarounds but would you mind using `releases_v1` instead of `releases.v1` ?

Reply:
That's ok.